src/Network/Ricochet/Util.hs

```diff
@@ -0,0 +1,29 @@
+module Network.Ricochet.Util
+  ( maybeResult'
+  , anyWord16
+  , joinWord8s
+  ) where
+
+import           Data.Attoparsec.ByteString
+import           Data.Bits                  (shiftL)
+import           Data.ByteString            (ByteString ())
+import           GHC.Word
+
+-- | Joins two Word8s into a single Word16
+joinWord8s :: (Word8, Word8) -> Word16
```

Comment:
Any particular reason why its not `joinWord8s :: Word8 -> Word8 -> Word16`?
Because you could write anyWord16 as `liftM2 joinWord8s anyWord8 anyWord8`, which would be more concise.